Source/WebCore/css/process-css-pseudo-selectors.py

```diff
@@ -435,37 +435,37 @@ def prefix_value(prefix, value):
     def write_parsing_function_definitions_for_pseudo_class(self, writer):
         longest_keyword_length = len(max(self.mapping, key=len))
         writer.write_block("""
-        static inline const SelectorPseudoClassOrCompatibilityPseudoElementEntry* findPseudoClassAndCompatibilityElementName(const LChar* characters, unsigned length)
+        static inline const SelectorPseudoClassOrCompatibilityPseudoElementEntry* findPseudoClassAndCompatibilityElementName(std::span<const LChar> characters)
         {
-            return SelectorPseudoClassAndCompatibilityElementMapHash::in_word_set(reinterpret_cast<const char*>(characters), length);
+            return SelectorPseudoClassAndCompatibilityElementMapHash::in_word_set(reinterpret_cast<const char*>(characters.data()), characters.size());
         }""")
 
         writer.write_block(f"""
-        static inline const SelectorPseudoClassOrCompatibilityPseudoElementEntry* findPseudoClassAndCompatibilityElementName(const UChar* characters, unsigned length)
+        static inline const SelectorPseudoClassOrCompatibilityPseudoElementEntry* findPseudoClassAndCompatibilityElementName(std::span<const UChar> characters)
         {{
             constexpr unsigned maxKeywordLength = {longest_keyword_length};
-            LChar buffer[maxKeywordLength];
-            if (length > maxKeywordLength)
+            std::array<LChar, maxKeywordLength> buffer;
+            if (characters.size() > maxKeywordLength)
                 return nullptr;
 
-            for (unsigned i = 0; i < length; ++i) {{
+            for (size_t i = 0; i < characters.size(); ++i) {{
                 UChar character = characters[i];
                 if (!isLatin1(character))
                     return nullptr;
 
                 buffer[i] = static_cast<LChar>(character);
             }}
-            return findPseudoClassAndCompatibilityElementName(buffer, length);
+            return findPseudoClassAndCompatibilityElementName(std::span {{ buffer }}.first(characters.size()));
```

Comment:
It’s amazing how tiny style things can catch my eye. I like `{{ }}`, maybe even prefer it, but we typically do `{ { } }` instead.

Reply:
Not sure `{. {`. The double curly bracket is purely due to python escaping.

Reply:
I could use `std::span(buffer)` to avoid python escaping if you'd prefer.

Reply:
It’s fine. I don’t know what python escaping is.

Reply:
`{foo}` is used for formatting strings in python, kinda like "%s". To clarify that we want to use '{' in the string instead of formatting, we need to double the curly brackets: `{{`.

https://stackoverflow.com/questions/5466451/how-do-i-escape-curly-brace-characters-in-a-string-while-using-format-or

Reply:
Oh, I didn’t realize this was in a python script. I thought it was a `.cpp` file. Wow, I missed a lot!